internal/core/metadata/application/device.go

```diff
@@ -237,3 +265,37 @@ func DevicesByProfileName(offset int, limit int, profileName string, dic *di.Con
 	}
 	return devices, totalCount, nil
 }
+
+var noMessagingClientError = goErrors.New("unable to publish Device System Event: MessageBus Client not available")
+
+func publishDeviceSystemEvent(action string, d models.Device, lc logger.LoggingClient, dic *di.Container) error {
+	device := dtos.FromDeviceModelToDTO(d)
+	systemEvent := dtos.NewSystemEvent(common.DeviceSystemEventType, action, common.CoreMetaDataServiceKey, device.ServiceName, nil, device)
+
+	messagingClient := bootstrapContainer.MessagingClientFrom(dic.Get)
+	if messagingClient == nil {
+		return noMessagingClientError
+	}
+
+	publishTopic := fmt.Sprintf("%s/%s/%s/%s/%s/%s",
+		common.SystemEventsPublishTopicPrefix,
+		systemEvent.Source,
+		systemEvent.Type,
+		systemEvent.Action,
+		systemEvent.Owner,
+		device.ProfileName)
+
+	payload, _ := json.Marshal(systemEvent)
+
+	envelope := types.NewMessageEnvelope(payload, context.Background())
+	envelope.CorrelationID = uuid.NewString()
```

Comment:
the correlation id should retrieve from Context, and it will be the same one as the changing device request for debugging